internal/widget/sortedTable.go

```diff
@@ -0,0 +1,254 @@
+package widget
+
+import (
+	"sort"
+
+	"github.com/gdamore/tcell/v2"
+	"github.com/rivo/tview"
+)
+
+type SortableRow interface {
+	Strings() []string
+	LessThan(Other SortableRow, column int) bool
+}
+
+type tableRow struct {
+	key  string
+	data SortableRow
+}
+
+// SortedTable is a wrapper around tview.Table which provides sortable column headers.  Rows are
+// identified by a key rather than by index.
+type SortedTable struct {
+	table       *tview.Table
+	values      []tableRow
+	curRow      int
+	sortColumn  int
+	sortReverse bool
+
+	columnAlign map[int]int
+
+	selectionChangedFunc func(key string)
+}
+
+// Mostly implement tview.Primitive with proxies
+func (st *SortedTable) Draw(screen tcell.Screen) {
+	st.Redraw()
+	st.table.Draw(screen)
+}
+
+func (st *SortedTable) GetRect() (int, int, int, int) {
+	return st.table.GetRect()
+}
+
+func (st *SortedTable) SetRect(x, y, width, height int) {
+	st.table.SetRect(x, y, width, height)
+}
+
+func (st *SortedTable) InputHandler() func(event *tcell.EventKey, setFocus func(p tview.Primitive)) {
+	return st.table.InputHandler()
+}
+
+func (st *SortedTable) Focus(delegate func(p tview.Primitive)) {
+	st.table.Focus(delegate)
+}
+
+func (st *SortedTable) HasFocus() bool {
+	return st.table.HasFocus()
+}
+
+func (st *SortedTable) Blur() {
+	st.table.Blur()
+}
+
+func (st *SortedTable) MouseHandler() func(action tview.MouseAction, event *tcell.EventMouse, setFocus func(p tview.Primitive)) (consumed bool, capture tview.Primitive) {
```

Comment:
This is a bit of a mess because tview doesn't use types for their funcs, but the idea is that any time `st.table` is referenced, it gets transparently rewritten as `st`, otherwise the underlying Table is captured, and we don't receive events.